data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -87,16 +90,20 @@ public OpenSearchSink(final PluginSetting pluginSetting) {
     this.routingField = openSearchSinkConfig.getIndexConfiguration().getRoutingField();
     this.action = openSearchSinkConfig.getIndexConfiguration().getAction();
     this.indexManagerFactory = new IndexManagerFactory();
-
+    this.initialized = false;
+    this.lock = new ReentrantLock(true);
     try {
-      initialize();
-    } catch (final IOException e) {
-      this.shutdown();
-      throw new RuntimeException(e.getMessage(), e);
+        doInitialize();
+    } catch (IOException e) {
+        LOG.warn("Failed to initialize OpenSearch sink " + e.getMessage());
+    } catch (Exception e) {
+        this.shutdown();
```

Comment:
Why do you shutdown this sink here? I think we want to keep trying right?

Reply:
Not in this case. This is not case where it cannot be retried. When it is possible to retry, we don't throw exception (like in case of `catch (IOException e) {`

Reply:
We are catching a generic `Exception`. How do we know that it is not retryable?

Reply:
This is throwing for a generic `Exception`. This may or may not be retryable. I think you can shutdown on the `OpenSearchSinkException`. But, any other exception is probably too much.

Reply:
Only IOExceptions and very few OpenSearchExceptions are retryable as per this check
```
public boolean canRetry(final Exception e) {
        return (e instanceof IOException ||
                (e instanceof OpenSearchException &&
                        !NON_RETRY_STATUS.contains(((OpenSearchException) e).status().getStatus())));
    }
```

Reply:
I think the problem  here is that we still calling `doInitialize` in the constructor. I think if you remove this and then add the exception handling into `doInitialize` it would work.

```
public void doInitialize() {
  try {
    ...
  } catch(OpenSearchSinkException ex) {
     shutdown();
  } catch(Exception ex) { 
     // Let it retry.
     LOG.warn(...)
  }
}
```